cmd/src/actions_exec_backend_runner.go

```diff
@@ -47,6 +47,7 @@ func (x *actionExecutor) do(ctx context.Context, repo ActionRepo) (err error) {
 			status := ActionRepoStatus{Cached: true, Patch: result}
 			x.updateRepoStatus(repo, status)
 			x.logger.RepoCacheHit(repo, status.Patch != PatchInput{})
+			x.logger.progress.IncStepsComplete(int64(len(x.action.Steps)))
```

Comment:
Why not do that inside `x.logger.RepoCacheHit`? It seems a bit unnecessary for `actionExecutor` needing to know about the `x.logger.progress`.

Reply:
That also applies to the `x.logger.progress.SetTotalSteps`/`x.logger.progress.IncJobs()`/... calls above. I think _how_ that's reported and tracked should be concern of the logger and hidden from `actionExecutor`